tests/test_sdr.py

```diff
@@ -27,11 +27,22 @@ def assert_expected_results_in_vector(expected_results, vector_path):
     watershed_results_layer = None
     watershed_results_feature = None
     for key in expected_results:
-        numpy.testing.assert_approx_equal(
-            actual_results[key], expected_results[key], significant=6,
-            err_msg=(
-                f'actual_results: {actual_results}, '
-                f'expected_results: {expected_results}'))
+        # numpy.testing.assert_almost_equal(
+        #     expected_results[key], actual_results[key], decimal=6)
+
+        # In order to pass with GDAL<2.3 and GDAL>2.3:
+        # asserting equality to 5 significant figures instead of 6 decimal
+        # places. GDAL 2.3 introduced new warping behavior yielding different
+        # pixel values when also using a smoothing interpolation method.
+        # Surprisingly, these differences are not washed out by an
+        # aggregation such as zonal statistics.

```

Comment:
Since we're merging this into `release/3.9`, which requires `GDAL>3`, I think we can safely remove this comment.

And on a related note, since we no longer need to accommodate two versions of GDAL that have different warping behaviors (leading to small but notable numerical differences), is it reasonable to bump down the tolerance back to 6 significant digits?  Or maybe you're already doing this in the `assert_allclose` call, just below this?

Reply:
Good point! Decreased rtol to 0.000001